pkg/querier/codec/protobuf_codec_test.go

```diff
@@ -449,3 +455,311 @@ func TestProtobufCodec_Encode(t *testing.T) {
+							Floats: []promql.FPoint{
+								{T: 12, F: 3.4},
+								{T: 56, F: 7.8},
+							},
+						},
+						{
+							Metric: labels.FromStrings("name2", "value2"),
+							Floats: []promql.FPoint{
+								{T: 12, F: 3.4},
+								{T: 56, F: 7.8},
+							},
+						},
+					},
+				},
+			},
+			expectedUnmarshalResp: tripperware.PrometheusResponse{
+				Status: "success",
+				Data: tripperware.PrometheusData{
+					ResultType: "matrix",
+					Result: tripperware.PrometheusQueryResult{
+						Result: &tripperware.PrometheusQueryResult_Matrix{
+							Matrix: &tripperware.Matrix{
+								SampleStreams: []tripperware.SampleStream{
+									{
+										Labels: cortexpb.FromLabelsToLabelAdapters(labels.FromStrings("name1", "value1")),
+										Samples: []cortexpb.Sample{
+											{
+												TimestampMs: 12,
+												Value:       3.4,
+											},
+											{
+												TimestampMs: 56,
+												Value:       7.8,
+											},
+										},
+									},
+									{
+										Labels: cortexpb.FromLabelsToLabelAdapters(labels.FromStrings("name2", "value2")),
+										Samples: []cortexpb.Sample{
+											{
+												TimestampMs: 12,
+												Value:       3.4,
+											},
+											{
+												TimestampMs: 56,
+												Value:       7.8,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "scalar",
+			resp: &v1.Response{
+				Status: "success",
+				Data: &v1.QueryData{
+					ResultType: "scalar",
+					Result:     promql.Scalar{T: 1000, V: 2},
+				},
+			},
+			expectedUnmarshalResp: tripperware.PrometheusResponse{
+				Status: "success",
+				Data: tripperware.PrometheusData{
+					ResultType: "scalar",
+					Result: tripperware.PrometheusQueryResult{
+						Result: &tripperware.PrometheusQueryResult_RawBytes{
+							RawBytes: []byte(`{"resultType":"scalar","result":[1,"2"]}`),
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "string",
+			resp: &v1.Response{
+				Status: "success",
+				Data: &v1.QueryData{
+					ResultType: "string",
+					Result:     promql.String{T: 1000, V: "2"},
+				},
+			},
+			expectedUnmarshalResp: tripperware.PrometheusResponse{
+				Status: "success",
+				Data: tripperware.PrometheusData{
+					ResultType: "string",
+					Result: tripperware.PrometheusQueryResult{
+						Result: &tripperware.PrometheusQueryResult_RawBytes{
+							RawBytes: []byte(`{"resultType":"string","result":[1,"2"]}`),
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			prometheusResponse, err := createPrometheusQueryResponse(test.resp, test.cortexInternal)
+			require.NoError(t, err)
+
+			b, err := proto.Marshal(prometheusResponse)
+			require.NoError(t, err)
+
+			var resp tripperware.PrometheusResponse
+			err = proto.Unmarshal(b, &resp)
+			require.NoError(t, err)
+
+			require.Equal(t, test.expectedUnmarshalResp, resp)
```

Comment:
I am trying to understand this unit test. What's the main difference of this and the existing `TestProtobufCodec_Encode`?

I see we added `proto.Unmarshal` here while `TestProtobufCodec_Encode` compares the serialized results directly. Do we really need to test the protobuf serialization/deserialization for `tripperware.PrometheusResponse` here? Trying to reduce LoC as much as possible.

Reply:
yeah, it seems to be not necessary. let's remove it.